setupServers.js

```diff
@@ -0,0 +1,157 @@
+import * as fs from 'fs';
+import restify from 'restify';
+import {setupRoutes} from './routes';
+const {dialog} = require('electron');
+import sudo from 'electron-sudo';
+import {replace} from 'ramda';
+
+const httpsMessage = 'This application will establish an encrypted link ' +
+    'between the connector application and plotly 2.0 client. In order to ' +
+    'provide that communication tunnel, a new private key for your device ' +
+    'will be generated using the \'openssl\' command which requires ' +
+    'administrator\'s password.';
+
+// asking for sudo more than once, but show a dialog window only once
+let messageShown = false;
+const showSudoMessage = () => {
+    if (!messageShown) {
+        messageShown = true;
+        dialog.showMessageBox(
+            {type: 'info', buttons: ['OK'], message: httpsMessage}
+        );
+    }
+};
+
+const setupSecureRestifyServer = (
+    {keyFile, csrFile, sequelizeManager,
+    serverMessageReceive, mainWindow, OPTIONS}
+) => {
+
+    const https_options = {
+        key: fs.readFileSync(keyFile),
+        certificate: fs.readFileSync(csrFile)
+    };
+
+    const server = restify.createServer(https_options);
+    server.use(restify.queryParser());
+    server.use(restify.CORS({
+        origins: ['*'],
```

Comment:
At some point we'll need to lock this down to just `https://plot.ly` and a configurable plotly enterprise URL domain.